### PR TITLE
feat: allow use of server.name as server_id argument 

### PR DESCRIPTION
**Pull request by nats**
This PR modifies parser for operations that accept an integer server_id path parameter to accept a string command-line argument. For example, if account has a server named `foo-bar.bnr.la` here are some  example commands that are now valid:

```
bl server get foo-bar.bnr.la
bl server delete foo-bar.bnr.la
bl server console foo-bar.bnr.la
bl server action reboot foo-bar.bnr.la
```

During parsing, if the command-line argument provided for `server_id` cannot be converted to integer (i.e. `int(argument)` raises ValueError) then the argument is assumed to be the name of a server. The `server list` operation is used to fetch all servers, and the result filtered to find the integer ID of the server with `name == argument`.

If there is no such server, the parser displays an error:

```
error: SERVER_ID: cound not find 'foo-bar.bnr.la'
```

### CODE REVIEW

**Comment by nats** [comment]
You currently just have to 'know' this is supported, since the displayed help for `server_id` is pulled from the OpenAPI spec:

```
$ grep '"server_id"' -A 4 src/binarylane/console/commands/api/* | grep -oP "description=.*" | sort | uniq -c
    1 description="""The ID of the server for which actions should be listed.""",
      1 description="""The ID of the server for which advanced features should be listed.""",
      1 description="""The ID of the server for which backups should be listed.""",
      1 description="""The ID of the server for which console URLs will be fetched.""",
      1 description="""The ID of the server for which firewall rules should be listed.""",
      1 description="""The ID of the server for which kernels should be listed.""",
      1 description="""The ID of the server for which snapshots should be listed.""",
      1 description="""The ID of the server for which software should be fetched.""",
      1 description="""The ID of the server for which the action should be fetched.""",
      1 description="""The ID of the server for which the backup is to be uploaded.""",
      1 description="""The ID of the server for which threshold alerts should be fetched.""",
      1 description="""The ID of the server for which userdata should be fetched.""",
     41 description="""The ID of the server on which the action should be performed.""",
      1 description="""The ID of the server to be cancelled.""",
      1 description="""The ID of the server to fetch.""",
      6 description="""The target server id.""",
```

@david-pershouse not sure how best to address that, any thoughts?

**Review by david-pershouse — CHANGES_REQUESTED** [review]
The _context passed to create_client in the lookup operations doesn't have the correct config. If I pass `--context dev` on the command line, the lookup runner ignores it and hits api.binarylane.com to resolve the name, then actual requested command executes against the dev context as expected.

**Review by david-pershouse — COMMENTED** [review]
Nice clean solution, confirms it fixes it here

File: templates/endpoint_module/create_mapping.jinja [hunk 1]
```diff
@@ -5,12 +5,30 @@
...
         {% for property in endpoint.path_parameters.values() %}
+        {# Check if parameter has a lookup command, and define handler if it does #}
+        {% set lookup_endpoint = lookup_parameters.get(property.name) %}
+        {% if lookup_endpoint %}
+        {% set lookup_entity = lookup_endpoint.data['x-cli-entity'] %}
+        def lookup_{{ property.python_name }}(ref: str) -> Union[None, {{ property.get_type_string() }}]:
+            return api_{{ python_identifier(lookup_endpoint.name) }}.Command(self._context).lookup(ref)
+
+        {% endif %}
+        {% if lookup_endpoint %}
```
[the remark this answers is not in the document]
Comment (nats) [reply templates/endpoint_module/create_mapping.jinja]: just a derp it seems while doing the id or name stuff.

**Review by david-pershouse — APPROVED** [review]

**Pull request merged**